### PR TITLE
Gitpod and ddev support

### LINKED ISSUE #100: Support for Gitpod
For the last few months, I've been using Gitpod as an online dev environment for LocalGov Drupal

This is a useful tool because:

* You can set up a working version of LocalGov Drupal in a few clicks with no local dependencies
* You can explore the content editing experience, as well as the public-facing site
* It works even if you have no local admin rights or similar (which is the case for many of us in councils).

At the moment I'm using ECC's Github to store this, but having the required files in this repository would mean that we don't have to update it in two places, and it would be more visible to other councils.

### PR DESCRIPTION
**Pull request by keelanfh**
Closes #100 

@finnlewis might be good to look at this if you are looking at #92? This is the ddev config file that I have set up for Gitpod and have been using for a while. We don't have to use exactly this config for ddev, but happy to test anything you're working on to see that it works in Gitpod too.

### CODE REVIEW

**Comment by keelanfh** [comment]
Couple of things to think about: 

* I'm not sure how useful the VS Code extensions in here are, we could remove them (or at least make sure they work, I get some warnings in Gitpod in the current config)
* When I run in Gitpod I don't seem to see stack traces. I don't know if there's an issue with how the settings.php file is being applied in this method?

**Comment by finnlewis** [comment]
DDEV works locally and feels really nice! Kind of a drop in replacement for Lando. Demo site installs well and it all feels pretty snappy. (I'd like to benchmark DDEV against Lando actually),

I thought I'd change the default hostname from localgov-project to localgov to be consistent with Lando config, is the OK @keelanfh ?

I'll take a look at it running in Gitpod too.

**Comment by finnlewis** [comment]
I would LOVE to get xdebug working in Gitpod... so the [xdebug vscode extension](https://github.com/localgovdrupal/localgov_project/pull/102/files#diff-370a022e48cb18faf98122794ffc5ce775b2606b09a9d1f80b71333425ec078eR18) is good to have.

**Comment by keelanfh** [comment]
Yep @finnlewis that's fine. Thanks for updating the drupalpod image too, that was on my list! :D

**Comment by finnlewis** [comment]
I'm getting this error now... is it to do with the change of image, or just limitations on gitpod space? 

image

**Comment by keelanfh** [comment]
I'm getting the same error. The error isn't occurring when I go back to https://github.com/localgovdrupal/localgov_project/pull/102/commits/a6482c029f3221e6dabfa0f634b325052601404e, so it does look like it's to do with the new image.

**Comment by keelanfh** [comment]
If you have Docker running locally, want to try just running `docker pull drupalpod/drupalpod-gitpod-base/20221216` to see if that works?

**Comment by keelanfh** [comment]
Seems to be working fine for me with the latest image, after my last change!

**Comment by finnlewis** [comment]
Doh! Thanks @keelanfh !

**Comment by finnlewis** [comment]
I have debugging working in Gitpod now, which is nice :)

Any idea how we add a default launch.json config to the Gitpod setup?

It needs to contain this: 

```
{
    "configurations": [
        {
            "name": "Listen for Xdebug",
            "type": "php",
            "request": "launch",
            "port": 9003,
            "pathMappings": {
                "/var/www/html": "${workspaceRoot}"
              }
        }
    ]
}

**Comment by keelanfh** [comment]
Try that? There was something similar in the original repo - https://github.com/shaal/ddev-gitpod/blob/main/.theia/launch.json - but I removed it as I was slimming it down. But if it's useful, let's add it back in!

**Comment by finnlewis** [comment]
Yay! xdebug working out of the box! I am sooooo happy! 

image

**Comment by finnlewis** [comment]
It would be great to get some other people to test this.

I do think we should bring it into the project and aim to support it, it is great for testing and demonstrations and could be a game changer for some councils and developers. 

Including default configuration in .vscode for ddev might be mildly annoying for someone using vscode and lando, but I tend to have my workspace in a folder above the project folder so it doesn't conflict for me. It is easy enough to document the differences and change the benefits probably outweigh the annoyance. 

@andybroomfield @Adnan-cds @ekes @stephen-cox penny for your thoughts? 


To test this go to: https://gitpod.io/#https://github.com/localgovdrupal/localgov_project/tree/feature/100-gitpod and spin it up!

**Comment by finnlewis** [comment]
Note: 
 - when flipping between Lando and DDEV the settings.php needs to change at present. 
 - we might be able to include an option based on environment variables to detect lando or ddev

File: assets/composer/settings.php [hunk 1]
```diff
@@ -754,6 +754,12 @@
 if (file_exists($app_root . '/' . $site_path . '/settings.local.php')) {
   include $app_root . '/' . $site_path . '/settings.local.php';
 }
-elseif (file_exists($app_root . '/' . $site_path . '/settings.lando.php')) {
+elseif (getenv('LANDO_SERVICE_NAME') == 'appserver' && file_exists($app_root . '/' . $site_path . '/settings.lando.php')) {
```
Comment (ekes) [inline assets/composer/settings.php:757]: I have a LANDO = ON, but can't find it in the documentation, which would feel more the equivalent of IS_DDEV_PROJECT. Still chances of anyone changing the prefix 'LANDO_' or the service name 'appserver' and not understanding this is minimal.

**Review by ekes — APPROVED** [review]
Looks fine, can't run ddev at the moment - tooooo many docker containers - but lando version is still working with this branch, happy if others have fired up ddev with it.

**Pull request merged**